diff --git a/repack/debian/changelog b/repack/debian/changelog
new file mode 100644
index 000000000..dde6aa6f8
--- /dev/null
+++ b/repack/debian/changelog
@@ -0,0 +1,13 @@
+stereolabs-zedx (0.4.9-MAX96712-L4T32.7.4) stable; urgency=low
+
+  * Bump to L4T 32.7
+  * ZED X multi-carrier board support
+
+ -- Stereolabs Inc <support@stereolabs.com>  Thu, 6 Apr 2023 12:11:22 +0200
+
+nvidia-l4t-kernel-dtbs (4.9.337-tegra-32.7.4) stable; urgency=low
+
+  * Bump to L4T 32.7
+  * Refer to Package Release Notes for details: http://developer.nvidia.com/jetson
+
+ -- NVIDIA Corporation <linux-tegra-bugs@nvidia.com>  Sat, 19 Feb 2022 09:03:44 -0800
diff --git a/repack/debian/control b/repack/debian/control
new file mode 100644
index 000000000..ad783c213
--- /dev/null
+++ b/repack/debian/control
@@ -0,0 +1,14 @@
+Source: stereolabs-zedx
+Section: kernel
+Priority: standard
+Maintainer: NVIDIA Corporation, Stereolabs Inc.
+Build-Depends: debhelper (>=9)
+Standards-Version: 3.9.6
+Homepage: www.stereolabs.com
+
+Package: stereolabs-zedx
+Architecture: arm64
+Pre-Depends: nvidia-l4t-core (>> 32.7-0), nvidia-l4t-core (<< 32.8-0),
+Depends: device-tree-compiler, nvidia-l4t-kernel (=4.9.337-tegra-32.7.4-20230608211515)
+Description: NVIDIA Kernel DTB Modified Package by Stereolabs
+X-Jetpack_Base: L4T32.7.4
