diff --color -Naur a/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3737-0000.dtsi b/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3737-0000.dtsi
--- a/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3737-0000.dtsi	2025-01-01 00:00:00.000000000 +0000
+++ b/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3737-0000.dtsi	2025-01-02 00:00:00.000000000 +0000
@@ -1,5 +1,9 @@
 // SPDX-License-Identifier: GPL-2.0-only
 // SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#include <dt-bindings/gpio/tegra234-gpio.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+
+//sl_placeholder
 
 / {
 	bus@0 {
diff --color -Naur a/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3768-0000.dtsi b/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3768-0000.dtsi
--- a/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3768-0000.dtsi	2025-01-01 00:00:00.000000000 +0000
+++ b/src/hardware/nvidia/t23x/nv-public/nv-platform/tegra234-p3768-0000.dtsi	2025-01-02 00:00:00.000000000 +0000
@@ -1,5 +1,9 @@
 // SPDX-License-Identifier: GPL-2.0-only
 // SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#include <dt-bindings/gpio/tegra234-gpio.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+
+//sl_placeholder
 
 / {
 	bus@0 {
--- a/src/kernel_src_build_env.sh	2025-01-01 00:00:00.000000000 +0000
+++ b/src/kernel_src_build_env.sh	2025-01-02 00:00:00.000000000 +0000
@@ -32,5 +32,5 @@
 # shellcheck disable=SC2034
 KERNEL_SRC_DIR="kernel-jammy-src"
 KERNEL_DEF_CONFIG="defconfig"
-OOT_SOURCE_LIST="nvethernetrm  nvgpu  nvidia-oot hwpm hardware nvdisplay kernel-devicetree"
+OOT_SOURCE_LIST="nvethernetrm  nvgpu  nvidia-oot hwpm hardware nvdisplay kernel-devicetree kernel/stereolabs"
 KERNEL_MODULAR_BUILD="y"
diff --color -Naur a/src/nvidia-oot/device-tree/platform/generic-dts/Makefile b/src/nvidia-oot/device-tree/platform/generic-dts/Makefile
--- a/src/nvidia-oot/device-tree/platform/generic-dts/Makefile	2025-01-01 00:00:00.000000000 +0000
+++ b/src/nvidia-oot/device-tree/platform/generic-dts/Makefile	2025-01-02 00:00:00.000000000 +0000
@@ -47,6 +47,11 @@
 # SOC independent common include
 DTC_INCLUDE += $(tegra-dtstree)/tegra/nv-public
 
+# Stereolabs dts includes
+DTC_INCLUDE += $(tegra-dtstree)/../stereolabs/AGX_Orin
+DTC_INCLUDE += $(tegra-dtstree)/../stereolabs/NANO_Orin
+DTC_INCLUDE += $(tegra-dtstree)/../stereolabs/NX_Orin
+
 # SOC T23X specific common include
 DTC_INCLUDE += $(tegra-dtstree)/t23x/nv-public/include/kernel
 DTC_INCLUDE += $(tegra-dtstree)/t23x/nv-public/include/nvidia-oot
--- a/src/Makefile	2025-01-01 00:00:00.000000000 +0000
+++ b/src/Makefile	2025-01-02 00:00:00.000000000 +0000
@@ -7,11 +7,12 @@
 MAKEFILE_DIR := $(abspath $(shell dirname $(lastword $(MAKEFILE_LIST))))
 NVIDIA_HEADERS ?= ${MAKEFILE_DIR}/out/nvidia-linux-header
 NVIDIA_CONFTEST ?= ${MAKEFILE_DIR}/out/nvidia-conftest
+SL_DRIVERS ?= ${MAKEFILE_DIR}/stereolabs/
 NVIDIA_DTS_BUILD_SCRIPTS ?= $(realpath ${MAKEFILE_DIR}/kernel-devicetree)
 
 ifneq ($(words $(subst :, ,$(MAKEFILE_DIR))), 1)
 $(error source directory cannot contain spaces or colons)
 endif
 
-NPROC ?= $(shell nproc)
+NPROC ?= $(shell echo $$(( $$(nproc) - 2 )))
 
@@ -25,12 +27,14 @@
 	@echo   "================================================================================"
 	@echo   "Usage:"
 	@echo   "   make modules          # to build NVIDIA OOT and display drivers"
+	@echo   "   make modules_oot      # to build SL OOT"
 	@echo   "   make dtbs             # to build NVIDIA DTBs"
 	@echo   "   make modules_install  # to install drivers to the INSTALL_MOD_PATH"
 	@echo   "   make clean            # to make clean driver sources"
 	@echo   "================================================================================"
 
-modules: hwpm nvidia-oot nvgpu nvidia-display
+modules: hwpm nvidia-oot stereolabs nvgpu nvidia-display
+modules_oot: stereolabs
 dtbs: nvidia-dtbs
 modules_install: hwpm nvidia-oot nvgpu nvidia-display-install
 clean: hwpm nvidia-oot nvgpu nvidia-display-clean nvidia-dtbs-clean
@@ -52,6 +56,7 @@
 endif
 
 hwpm: conftest
+ifneq ($(MAKECMDGOALS), modules_oot)
 	@if [ ! -d "$(MAKEFILE_DIR)/hwpm" ] ; then \
 		echo "Directory hwpm is not found, exiting.."; \
 		false; \
@@ -66,8 +71,10 @@
 		srctree.hwpm=$(MAKEFILE_DIR)/hwpm \
 		srctree.nvconftest=$(NVIDIA_CONFTEST) \
 		$(MAKECMDGOALS)
+endif
 
 nvidia-oot: conftest hwpm
+ifneq ($(MAKECMDGOALS), modules_oot)
 	@if [ ! -d "$(MAKEFILE_DIR)/nvidia-oot" ] ; then \
 		echo "Directory nvidia-oot is not found, exiting.."; \
 		false; \
@@ -84,6 +91,44 @@
 		srctree.nvconftest=$(NVIDIA_CONFTEST) \
 		KBUILD_EXTRA_SYMBOLS=$(MAKEFILE_DIR)/hwpm/drivers/tegra/hwpm/Module.symvers \
 		$(MAKECMDGOALS)
+endif
+
+stereolabs: nvidia-headers
+ifneq ($(MAKECMDGOALS), modules_oot)
+	@if [ ! -d "$(MAKEFILE_DIR)/stereolabs" ] ; then \
+		echo "Directory stereolabs is not found, exiting.."; \
+		false; \
+	fi
+	@echo   "================================================================================"
+	@echo   "make $(MAKECMDGOALS) - stereolabs ..."
+	@echo   "================================================================================"
+	$(MAKE) -j $(($(NPROC)-2)) ARCH=arm64 \
+		-C $(KERNEL_OUTPUT) -C $(NVIDIA_HEADERS) \
+		M=$(MAKEFILE_DIR)/stereolabs \
+		CONFIG_TEGRA_OOT_MODULE=m \
+		srctree.sl_drivers=$(SL_DRIVERS) \
+		KBUILD_EXTRA_SYMBOLS=$(MAKEFILE_DIR)/hwpm/drivers/tegra/hwpm/Module.symvers \
+		$(MAKECMDGOALS)
+else
+	@if [ ! -d "$(MAKEFILE_DIR)/stereolabs" ] ; then \
+		echo "Directory stereolabs is not found, exiting.."; \
+		false; \
+	fi
+	@if [ ! -d "$(NVIDIA_HEADERS)" ] ; then \
+		echo "Kernel headers not found, exiting.. You must compile all the sources at least once!"; \
+		false; \
+	fi
+	@echo   "================================================================================"
+	@echo   "make $(MAKECMDGOALS) - stereolabs ..."
+	@echo   "================================================================================"
+	$(MAKE) -j $(($(NPROC)-2)) ARCH=arm64 \
+		-C $(KERNEL_HEADERS) -C $(NVIDIA_HEADERS) \
+		M=$(MAKEFILE_DIR)/stereolabs \
+		CONFIG_TEGRA_OOT_MODULE=m \
+		srctree.sl_drivers=$(SL_DRIVERS) \
+		KBUILD_EXTRA_SYMBOLS=$(MAKEFILE_DIR)/hwpm/drivers/tegra/hwpm/Module.symvers \
+		modules
+endif
 
 nvgpu: conftest nvidia-oot
 	if [ ! -d "$(MAKEFILE_DIR)/nvgpu" ] ; then \
@@ -122,6 +167,7 @@
 
 
 nvidia-headers: nvidia-oot
+ifneq ($(MAKECMDGOALS), modules_oot)
 	mkdir -p $(NVIDIA_HEADERS)
 	cp -LR $(KERNEL_HEADERS)/* $(NVIDIA_HEADERS)
 	if [ "$(KERNEL_HEADERS)" != "$(KERNEL_OUTPUT)" ] ; then \
@@ -133,6 +179,7 @@
 	cp -LR $(MAKEFILE_DIR)/nvidia-oot/include/* $(NVIDIA_HEADERS)/include/
 	cat $(KERNEL_OUTPUT)/Module.symvers $(MAKEFILE_DIR)/nvidia-oot/Module.symvers > \
 		$(NVIDIA_HEADERS)/Module.symvers
+endif
 
 
 nvidia-display: nvidia-headers
@@ -155,7 +202,7 @@
 	@echo   "================================================================================"
 	@echo   "make $(MAKECMDGOALS) - nvidia-display ..."
 	@echo   "================================================================================"
-	$(MAKE) -C $(NVIDIA_HEADERS) INSTALL_MOD_DIR=updates/opensrc-disp \
+	$(MAKE) -j $(($(NPROC)-2)) -C $(NVIDIA_HEADERS) INSTALL_MOD_DIR=updates/opensrc-disp \
 		M=$(MAKEFILE_DIR)/nvdisplay/kernel-open modules_install
 
 nvidia-display-clean:
@@ -178,7 +225,7 @@
 	objtree=$(KERNEL_OUTPUT) \
 	oottree=$(NVIDIA_DTS_BUILD_SCRIPTS) \
 	HOSTCC=gcc \
-	$(MAKE) -f $(NVIDIA_DTS_BUILD_SCRIPTS)/scripts/Makefile.build \
+	$(MAKE) -j $(($(NPROC)-2)) -f $(NVIDIA_DTS_BUILD_SCRIPTS)/scripts/Makefile.build \
 		obj=$(NVIDIA_DTS_BUILD_SCRIPTS)/generic-dts \
 		dtbs
 	@echo   "================================================================================"
--- a/src/nvbuild.sh	2025-01-01 00:00:00.000000000 +0000
+++ b/src/nvbuild.sh	2025-01-02 00:00:00.000000000 +0000
@@ -50,6 +50,7 @@
         -h              Displays this help
         -r              Enable RT Kernel
         -m              Build/install NVIDIA OOT modules only (skip kernel)
+        -mr             Build/install SL OOT set of modules only (skip kernel/display/nv-oot/...)
         -i              Perform install to INSTALL_MOD_PATH
         -o <outdir>     Creates kernel build output in <outdir>
 EOM
@@ -75,6 +76,10 @@
 				OOT_MODULES_ONLY=1
 				shift 1
 				;;
+			-mr)
+				REDUCED_OOT_MODULES_ONLY=1
+				shift 1
+				;;
 			-i)
 				DO_INSTALL=1
 				shift 1
@@ -88,7 +93,7 @@
 	done
 }
 
-# Function to kernel sources 
+# Function to kernel sources
 function sync_kernel_source {
 	# Check whether kernel source dir available
 	if [ ! -d "kernel/${KERNEL_SRC_DIR}" ]; then
@@ -98,11 +103,11 @@
 
 	# Sync kernel source directory to output dir
 	echo "Syncing kernel/${KERNEL_SRC_DIR}"
-	rsync -a --delete "kernel/${KERNEL_SRC_DIR}" "${KERNEL_OUT_DIR}/kernel/"
+	rsync -a "kernel/${KERNEL_SRC_DIR}" "${KERNEL_OUT_DIR}/kernel/"
 	cp -a kernel/Makefile "${KERNEL_OUT_DIR}/kernel/"
 }
 
-# Function to rsync oot modules sources 
+# Function to rsync oot modules sources
 function sync_oot_modules {
 	# Check whether all source directories all available from kernel source list
 	for list in ${OOT_SOURCE_LIST}
@@ -117,7 +122,7 @@
 	for list in ${OOT_SOURCE_LIST}
 	do
 		echo "Syncing ${list}"
-		rsync -a --delete "${list}" "${KERNEL_OUT_DIR}"
+		rsync -a "${list}" "${KERNEL_OUT_DIR}"
 	done
 	cp -a Makefile "${KERNEL_OUT_DIR}/"
 }
@@ -142,8 +147,12 @@
 	export KERNEL_SRC_DIR="${KERNEL_SRC_DIR}"
 	export KERNEL_DEF_CONFIG="${KERNEL_DEF_CONFIG}"
 
-	echo "Building kernel sources in ${kernel_out_dir}"
-	make -C "${kernel_out_dir}"
+	if [ "${REDUCED_OOT_MODULES_ONLY}" -eq 1 ]; then
+		echo "Skipping the kernel build, just testing it was compiled once"
+	else
+		echo "Building kernel sources in ${kernel_out_dir}"
+		make -C "${kernel_out_dir}"
+	fi
 
 	image="${kernel_out_dir}/${KERNEL_SRC_DIR}/arch/arm64/boot/Image"
 	if [ ! -f "${image}" ]; then
@@ -167,7 +176,13 @@
 
 	echo "Building NVIDIA OOT modules sources in ${KERNEL_OUT_DIR}"
 	echo "Using KERNEL_HEADERS from directory : ${KERNEL_HEADERS}"
-	make -C "${KERNEL_OUT_DIR}" modules
+	if [ "${REDUCED_OOT_MODULES_ONLY}" -eq 1 ]; then
+		make -C "${KERNEL_OUT_DIR}" modules_oot
+	else
+		make -C "${KERNEL_OUT_DIR}" modules
+	fi
+
+	touch "${KERNEL_OUT_DIR}"/.BUILD_ONCE
 
 	echo "Building NVIDIA DTBs"
 	make -C "${KERNEL_OUT_DIR}" dtbs
@@ -209,7 +224,7 @@
 			echo "Error: Path ${CROSS_COMPILE}gcc does not exist."
 			exit 1
 		fi
-		if [ "${OOT_MODULES_ONLY}" -eq 1 ]; then
+		if [[ "${OOT_MODULES_ONLY}" -eq 1 || "${REDUCED_OOT_MODULES_ONLY}" -eq 1 ]]; then
 			if [ -z "${KERNEL_HEADERS}" ]; then
 				echo "Error: Env variable KERNEL_HEADERS is not set!!"
 				exit 1
@@ -256,7 +271,7 @@
 		exit 1
 	fi
 
-	if [ "${OOT_MODULES_ONLY}" -eq 0 ]; then
+	if [[ "${OOT_MODULES_ONLY}" -eq 0 && "${REDUCED_OOT_MODULES_ONLY}" -eq 0 ]]; then
 		install_kernel
 	fi
 
@@ -270,7 +285,7 @@
 fi
 
 # Compile kernel sources for "arm64"
-if [ "${OOT_MODULES_ONLY}" -eq 0 ]; then
+if [[ "${OOT_MODULES_ONLY}" -eq 0 && "${REDUCED_OOT_MODULES_ONLY}" -eq 0 ]]; then
 	build_arm64_kernel_sources
 fi
 
